### ISSUE TITLE
ESP32 S3 Hardware and LCD connections

### ISSUE THREAD

**Issue by phodina** [mrfaptastic/ESP32-HUB75-MatrixPanel-DMA#551]
Hi,

I'm designing a controller board for a [backpack](https://gitlab.com/phodina/badboard) I recently bought which had let's say poor and proprietary implementation.

Normally I'd use STM32F4 with the FSMC interface to drive the RGB LED panel. However, as I'd like to connect to the controller over wireless (not sure what's better WiFi/Bluetooth) I have switch to ESP32.

I used mostly ESP32 C3 to play with the RISC-V core.

However, I never used the ESP32 and I could not find in the datasheet in the pinout nor LCD section anything about the pins that are available.

I checked the default pins and set accordingly the connections in the schematics.
https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/blob/master/src/platforms/esp32s3/esp32s3-default-pins.hpp
![image](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/2997905/eda79fad-1062-40a9-8aa1-eea0186e2302)

Also the suggested link for the hardware seems to be no longer valid.
![image](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/2997905/4c278b5a-bcbe-48a0-bf83-bcd0eb92de3d)

So is this the best setup?

Also is it possible to change pins for `G2` and `B2` as they are connected to the RTC crystal that I'd like to use for keeping precise time?

**Comment by mrfaptastic** [comment]
C series chips don't have the LCD module. Won't ever work. Unless you use PxMatrux library. Please read README what chips are supported.

**Comment by phodina** [comment]
> C series chips don't have the LCD module. Won't ever work. Unless you use PxMatrux library. Please read README what chips are supported.

Yes, that's why the selected MCU is **ESP32 S3**. That section is clear :)

> Various people have created PCBs for which one can simply connect an ESP32 to a PCB, and then the PCB to the HUB75 connector, such as:

 >   Brian Lough's [ESP32 I2S Matrix Shield](http://blough.ie/i2smat/)
 
 This link is no longer available.
 
What I miss is the description of the pins. In the ESP32 datasheet the pinout does not list which pins are used for LCD.
https://www.espressif.com/sites/default/files/documentation/esp32-s3_datasheet_en.pdf
![image](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/2997905/9e9cfb47-b3b3-4c1c-8077-e1c5c1991855)

Lastly can these pins be remapped?

**Comment by phodina** [comment]
I've been able to find the mapping for SPI RAM on the ESP32 S3 chip
https://api.riot-os.org/group__cpu__esp32__esp32s3.html
![image](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/2997905/0893bf45-9dcb-465b-a450-2c369432de95)

Though there's the speed limitation
> For the ESP32-S3 only, you can use SPIRAM/PSRAM to drive the HUB75 DMA buffer when using an ESP32-S3 with OCTAL SPI-RAM (PSTRAM) (i.e. ESP32 S3 N8R8 variant). However, due to bandwidth limitations, the maximum output frequency is limited to approx. 13Mhz, which will limit the real-world number of panels that can be chained without flicker. Please do not use PSRAM as the DMA buffer if using QUAD SPI (Q-SPI), as it's too slow.

**Comment by phodina** [comment]
In the ESP-IDF source code there's the definition for the pins but still does not document the LCD peripheral

https://github.com/espressif/esp-idf/blob/ee3a46dbc3aa8575db42af6afbe37acbc2280bbe/examples/peripherals/lcd/rgb_panel/main/rgb_lcd_example_main.c#L33

**Comment by toine512** [comment]
Chapter 6 of the ESP32-S3 TRM.
All pins not directly affected to special interfaces (USB, JTAG, SPI flash, SPI RAM) are usable. That means all pins left available to the GPIO matrix.

Default GPIO mapping used by the library is here : https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/blob/master/src/platforms/esp32s3/esp32s3-default-pins.hpp
If you need E you may use GPIO39 for example.

**Comment by mrfaptastic** [comment]
Default library pins for S3 are defined here:

https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/blob/master/src%2Fplatforms%2Fesp32s3%2Fesp32s3-default-pins.hpp

**Comment by phodina** [comment]
> ESP32-S3 TRM
> All pins not directly affected to special interfaces (USB, JTAG, SPI flash, SPI RAM) are usable. That means all pins left available to the GPIO matrix (not in "direct connection" mode).

Thanks. That's exactly what I was looking for.

**Comment by phodina** [comment]
> Default library pins for S3 are defined here:
> 
> https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/blob/master/src%2Fplatforms%2Fesp32s3%2Fesp32s3-default-pins.hpp

Thanks. For the definition of the pins but the there but the question was more broad as what other pins can be used :)  as I'd like some peripherals to be used and it would conflict with the default pins.
